### PR TITLE
pctpairprotons in Python

### PR DESCRIPTION
**Pull request by acoussat**
Reimplementation of `pctpairprotons` in Python in order to remove the dependency on ROOT. The current behavior is that the script is always copied to the build folder, and the C++ version of `pctpairprotons` is only compiled when `PCT_WITH_ROOT` is on. I checked on simple cases that the C++ and the Python versions produce the same outputs at the byte level.

The Python version adds a dependency on `pandas` to efficiently merge the two ROOT files. I couldn't find an easy way to reproduce the same behavior in pure NumPy. I did write a pure NumPy implementation of the same script, but it uses `for` loops pretty much like the C++ version does. I didn't include it in this PR, but it is accessible there: https://github.com/acoussat/PCT/blob/pctpairprotons-numpy/pctpairprotons_numpy.py. I didn't extensively compare the two implementations to check which one is faster, but I would guess that `pandas`' optimized routine is probably faster.

Let me know what you think!

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -63,18 +68,18 @@ MACRO (ADD_PCT_EXECUTABLE EXE_NAME)
...
   ADD_PCT_EXECUTABLE(pctpairprotonsLomaLinda pctpairprotonsLomaLinda.cxx)
   TARGET_LINK_LIBRARIES(pctpairprotonsLomaLinda ${ROOT_LIBRARIES})
   ADD_PCT_EXECUTABLE(pctpairprotonsLMU_IMPCT pctpairprotonsLMU_IMPCT.cxx)
   TARGET_LINK_LIBRARIES(pctpairprotonsLMU_IMPCT ${ROOT_LIBRARIES})
-ENDIF(NOT WIN32)
-ADD_PCT_EXECUTABLE(pctpaircuts pctpaircuts.cxx)
-TARGET_LINK_LIBRARIES(pctpaircuts ${ROOT_LIBRARIES} RooFit RooFitCore Minuit)
-ADD_PCT_EXECUTABLE(pctlluconverter pctlluconverter.cxx)
-TARGET_LINK_LIBRARIES(pctlluconverter ${ROOT_LIBRARIES})
+  ADD_PCT_EXECUTABLE(pctpaircuts pctpaircuts.cxx)
```
Comment (SimonRit) [inline CMakeLists.txt:78]: It's good to remove the root dependency but we can't entirely disable pctpaircuts. Would be nice to have a compilation define (e.g., `PCT_USE_ROOT`) to disable certain features in pctpaircuts.

Comment (acoussat) [reply CMakeLists.txt:78]: I have implemented such idea atop `master` here: https://github.com/acoussat/PCT/commit/5ae18f51d8830f36a8b98f1abe4a4c77030d380e. Should I open a new PR with this code or update this PR?

Comment (SimonRit) [reply CMakeLists.txt:78]: No, I think it's part of the same PR.
Nothing wrong with your code but I'd consider creating a `pctConfiguration.h.in` similar to [RTK's `rtkConfiguration.h.in`](https://github.com/RTKConsortium/RTK/blob/master/rtkConfiguration.h.in). What do you think?

**Comment by acoussat** [comment]
I just noticed that I did not re-implemented the optional sixth column with nuclear information. Should I also re-implement it or is it unnecessary at this stage?

**Comment by SimonRit** [comment]
Yes, please reimplement it if it's not a lot of work. We can consider a reimplementation later on but it would be nice to start from just a translation of the original tool. Thanks in advance!

**Comment by acoussat** [comment]
> Yes, please reimplement it if it's not a lot of work. We can consider a reimplementation later on but it would be nice to start from just a translation of the original tool. Thanks in advance!

Unfortunately, it's not as easy as it looks, as GATE 10 does not export branches `NuclearProcess` and `Order`. After discussing with @dsarrut, it looks to me that there is no straightforward solution, even if we modify GATE. We can discuss it more in details if you want, but @dsarrut suggests to postpone this feature or simply remove this part of the code. What do you think?

**Comment by acoussat** [comment]
I squashed all my previous changes into https://github.com/SimonRit/PCT/pull/5/commits/10f72b609ea4be96d4cbfc8a7b2cd3d1e2946114.

File: CMakeLists.txt [hunk 2]
```diff
@@ -21,14 +21,19 @@ FIND_PACKAGE(Gengetopt REQUIRED)
 find_package(ITK 5.0.0 REQUIRED COMPONENTS RTK)
 include(${ITK_USE_FILE})
 
-IF(NOT WIN32)
+IF(NOT ITK_USE_FFTWD OR NOT ITK_USE_FFTWF)
```
Comment (SimonRit) [inline CMakeLists.txt:24]: I'm not sure why we have that but it could be removed I think

File: CMakeLists.txt [hunk 3]
```diff
@@ -21,14 +21,19 @@ FIND_PACKAGE(Gengetopt REQUIRED)
...
+IF(PCT_WITH_ROOT AND NOT WIN32)
   FIND_PACKAGE(ROOT REQUIRED)
   INCLUDE_DIRECTORIES(${ROOT_INCLUDE_DIR})
   SET(LINK_DIRECTORIES ${ROOT_LIBRARY_DIR})
-ENDIF(NOT WIN32)
-
-IF(NOT ITK_USE_FFTWD OR NOT ITK_USE_FFTWF)
-    MESSAGE(FATAL_ERROR "You must recompile ITK with ITK_USE_FFTWF and ITK_USE_FFTWD. Then, recompile RTK first and pct second.")
+ELSE()
+  INCLUDE_DIRECTORIES(${CMAKE_CURRENT_SOURCE_DIR})
```
Comment (SimonRit) [inline CMakeLists.txt:36]: Why this?

File: CMakeLists.txt [hunk 4]
```diff
@@ -51,6 +56,10 @@ ENDIF()
 # such warning related to clhep/g4 system of units.
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-shadow")
 
+#=========================================================
+# Propagate cmake options in a header file
+configure_file(pctConfiguration.h.in pctConfiguration.h)
```
Comment (SimonRit) [inline CMakeLists.txt:61]: Maybe in the `PROJECT_BINARY_DIR`?

**Review by SimonRit — APPROVED** [review]
Just a few comments before merging. These comments can be part of a subsequent PR.

**Pull request merged**